### ISSUE TITLE
WARNING: Error saving Task history

### ISSUE THREAD

**Issue by EncoreHQ** [spotify/luigi#1132]
When Luigi is saving the task history it just throws this error:

```
WARNING: Error saving Task history
Traceback (most recent call last):
  File "/mnt/anaconda/lib/python2.7/site-packages/luigi/scheduler.py", line 972, in _update_task_history
    self._task_history.task_finished(task_id, successful)
  File "/mnt/anaconda/lib/python2.7/site-packages/luigi/db_task_history.py", line 90, in task_finished
    task = self._get_task(task_id, status=event_name)
  File "/mnt/anaconda/lib/python2.7/site-packages/luigi/db_task_history.py", line 104, in _get_task
    task = self.tasks[task_id] = task_history.Task(task_id, status, host)
  File "/mnt/anaconda/lib/python2.7/site-packages/luigi/task_history.py", line 38, in __init__
    self.task_family, self.parameters = task.id_to_name_and_params(task_id)
  File "/mnt/anaconda/lib/python2.7/site-packages/luigi/task.py", line 64, in id_to_name_and_params
    return luigi.tools.parse_task.id_to_name_and_params(task_id)
  File "/mnt/anaconda/lib/python2.7/site-packages/luigi/tools/parse_task.py", line 58, in id_to_name_and_params
    parsed = parser.parseString(task_id).asDict()
  File "/mnt/anaconda/lib/python2.7/site-packages/pyparsing.py", line 1125, in parseString
    raise exc
ParseException: Expected ")" (at char 28), (line:1, col:29)
```

Anyone have a clue on what is going on?

**Comment by erikbern** [comment]
you probably have a task with a parenthesis or something weird name

we *really* need to kill `id_to_name_and_params`. i've said it lots of times before. i should just go ahead and remove that function and all its references

**Comment by mfcabrera** [comment]
This was apparently fixed by #1309 so closing
